### PR TITLE
Php 22 create events

### PR DESCRIPTION
**Pull request by TimoStroehlein**

### CODE REVIEW

File: src/controllers/EventCreateController.php [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+        $this->view->pageTitle = "Create Event";
+        $this->view->isSuccess = isset($_GET["success"]);
+        $this->view->isError = isset($_GET["error"]);
+    }
+
+    /**
+     * Validate the teh data of the event, throws an error if something is wrong
+     * @param $data * Data of the event
+     */
+    private function validateData($data)
```
Comment (Haiyn) [inline src/controllers/EventCreateController.php]: Needs more validation:

- Assert that date/time not in past
- Assert that maximum attendees not 0
- Assert that text fields do not contain white space only (use sanitize filters when fetching $_POST data??)

File: src/models/User.php [hunk 2]
```diff
@@ -28,6 +28,18 @@ public static function newInstance()
         return new self();
     }
 
+    /*
+     * Searches the users table for a user with the passed user id
+     */
+    public function getUserById($user_id) {
+        $users = self::$_database->fetch(
+            "SELECT * from users WHERE username = :username",
```
Comment (Haiyn) [inline src/models/User.php]: This won't work, you're checking for username column instead of user_id

File: src/models/enums/Status.php [hunk 3]
```diff
@@ -2,10 +2,8 @@
 
 namespace models\enums;
 
-use SplEnum;
-
-class Status extends SplEnum
+class Status
```
Comment (Haiyn) [inline src/models/enums/Status.php]: Maybe make this class iterable with object iteration? https://www.php.net/manual/de/language.oop5.iterations.php
Could be in abstract parent class. That way, you can dynamically insert the options in the dropdown of the phtml file (class becomes expandable without front end changes needed)

File: src/models/enums/Visibility.php [hunk 4]
```diff
@@ -2,12 +2,8 @@
 
 namespace models\enums;
 
-use SplEnum;
-
-class Visibility extends SplEnum
+class Visibility
```
Comment (Haiyn) [inline src/models/enums/Visibility.php]: See Status class

File: src/views/event-create/event-create.phtml [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+            </div>
+            <div class="form-group">
+                <label for="event-create-time">Time</label>
+                <input id="event-create-time" type="time" name="time" class="form-control"><br/>
+            </div>
+            <div class="form-group">
+                <label for="event-create-visibility">Visibility*</label>
+                <select id="event-create-visibility" name="visibility" required class="form-control">
+                    <option selected><?php echo \models\enums\Visibility::$INVITE_ONLY ?></option>
+                    <option><?php echo \models\enums\Visibility::$PUBLIC ?></option>
```
Comment (Haiyn) [inline src/views/event-create/event-create.phtml]: See Enum classes to make this dynamic

File: src/controllers/EventCreateController.php [hunk 6]
```diff
@@ -0,0 +1,79 @@
...
+            $this->setSuccess("Event successfully created.");
+        }
+
+        $this->view->pageTitle = "Create Event";
+        $this->view->isSuccess = isset($_GET["success"]);
+        $this->view->isError = isset($_GET["error"]);
+    }
+
+    /**
+     * Validate the teh data of the event, throws an error if something is wrong
```
Comment (Haiyn) [inline src/controllers/EventCreateController.php]: lol

File: src/views/event-create/event-create.phtml [hunk 7]
```diff
@@ -0,0 +1,73 @@
...
+                <label for="event-create-location">Location</label>
+                <input id="event-create-location" type="text" name="location" class="form-control"><br/>
+            </div>
+            <div class="form-group">
+                <label for="event-create-date">Date*</label>
+                <input id="event-create-date" name="date" placeholder="yyyy-mm-dd" required class="form-control datepicker"><br/>
+            </div>
+            <div class="form-group">
+                <label for="event-create-time">Time</label>
+                <input id="event-create-time" type="time" name="time" class="form-control"><br/>
```
Comment (Haiyn) [inline src/views/event-create/event-create.phtml]: Is this changable to 24h clock instead of 12h clock (AM/PM)? Postgres also uses 24h

Comment (TimoStroehlein) [reply src/views/event-create/event-create.phtml]: The time format depends on the browser and system, so I couldn't find a good solution to fix this directly. So I instead changed the type to text and created a pattern to validate the time, let me know if that's better.

**Pull request merged**